src/lib/Components/Artist/Header.tsx

```diff
@@ -46,25 +51,32 @@ class Header extends React.Component<HeaderProps, State> {
     })
   }
 
-  handleFollowChange = () => {
+  @track({ action: "press follow/unfollow button" })
+  handleFollowChange() {
     const newFollowersCount = this.state.following ? this.state.followersCount - 1 : this.state.followersCount + 1
     ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, this.props.artist._id, (error, following) => {
       if (error) {
         console.error(error)
       } else {
-        Events.postEvent(this, {
-          name: following ? "Follow artist" : "Unfollow artist",
-          artist_id: this.props.artist._id,
-          artist_slug: this.props.artist.id,
-          // TODO At some point, this component might be on other screens.
-          source_screen: "artist page",
-        })
+        this.successfulFollowChange()
       }
       this.setState({ following, followersCount: newFollowersCount })
     })
     this.setState({ following: !this.state.following, followersCount: newFollowersCount })
   }
 
+  // currently you can't get state yet, but leaving this in as desired usage
+  @track((props, state) => ({ name: state.following, artist_id: props.artist._id, artist_slug: props.artist.id }))
```

Comment:
Btw, the value you’re passing as `name` isn’t the same as the original code, it should be a string.

Additionally, I’m now realising that this would be a non-issue by using the `Promise` pattern we discussed; that is, to send the status string based on the outcome of the API call.